### PR TITLE
docs: Simplify api build

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: docs: Simplify api build
Additional Description:

This fixes the refs in protos and removes the sed operations for build

The benefits are:

- less indirection in proto refs
- (small) speedup of build
- simplify docs build
- simplify reuse of proto comments (eg generating jsonschemas)

Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by htuch** [comment]
@phlax can you suggest how to review this PR? It's kind of huge. What are the key files to look at?

File: docs/build.sh [hunk 1]
```diff
@@ -132,11 +132,6 @@ function generate_api_rst() {
 
 generate_api_rst v3
 
-# Fixup anchors and references in v3 so they form a distinct namespace.
-# TODO(htuch): Do this in protodoc generation in the future.
-find "${GENERATED_RST_DIR}"/api-v3 -name "*.rst" -print0 | xargs -0 sed -i -e "s#envoy_api_#envoy_v3_api_#g"
-find "${GENERATED_RST_DIR}"/api-v3 -name "*.rst" -print0 | xargs -0 sed -i -e "s#config_resource_monitors#v3_config_resource_monitors#g"
-
```
Comment (phlax) [inline docs/build.sh:139]: this is the main change - ie not mangling the refs after the fact. They are kinda inconsistent in that in the proto files they sometimes do/not use the full `api_v3_etc` 

also the less mangling the better - eg the jsonschema wants to consume these descriptions too to turn them into markdown links

Comment (phlax) [reply docs/build.sh:139]: the second of these is redundant now anyway

File: tools/protodoc/protodoc.py [hunk 2]
```diff
@@ -432,27 +432,27 @@ def strip_leading_space(s):
 
 def file_cross_ref_label(msg_name):
     """File cross reference label."""
-    return 'envoy_api_file_%s' % msg_name
+    return 'envoy_v3_api_file_%s' % msg_name
```
Comment (phlax) [inline tools/protodoc/protodoc.py:435]: this is basically your comment above i think - ie doing it in protodoc

Comment (phlax) [reply tools/protodoc/protodoc.py:435]: well, at least one interpretation of doing so these need to match the links essentially

**Comment by phlax** [comment]
Another reason this is important is that it makes the source code links == the rendered rst source links

this is probably the reason that there is a mixture of `_api_` and  `_api_v3_` is that - at least how i work - i copypasta the links from the rendered `.rst` source

this not being the case makes it confusing for people editing links in api or docs (that link to apis) as you must know about the mangling to know that they essentially going to render the same links

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**